### PR TITLE
add page to show all releases and their downloads

### PR DESCRIPTION
**Pull request by benesch**
This commit adds a page to display all releases and their downloads,
including binaries for Linux, Mac, and Windows (where available), and
source tarballs (where available). The sidebar is adjusted to display
only the latest version and a link to this page. The "old versions" page
is removed in favor of the new all-releases page.

Additionally adjusts the CSS to use the source archive icon consistently
across this page and the individual release notes pages.

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/1371)

### CODE REVIEW

**Comment by benesch** [comment]
Here's a screenshot:

screenshot 2017-05-04 18 53 55

I expect we'll replace the "Latest" badge with "Recommended" or "Stable" once 1.0 lands.

**Comment by benesch** [comment]
Also, @jseldess, I decided to stick with my icon because I think it's more evocative (and standard) than the generic down arrow. Let me know if you feel otherwise! This commit also adjusts the CSS for the button on the release page, though you can't see the effect because those buttons are added in the v1.0-rc.2 PR.

File: releases.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    vertical-align: middle;
+}
+</style>
+
+After downloading your desired release, learn how to [Install CockroachDB](install-cockroachdb.html).
+
+<table class="release-table">
+<thead>
+<tr>
+  <td>Version</td>
```
Comment (sploiselle) [inline releases.md]: <td>Version &amp; Release Notes</td> just so it's clear what's at the other end of the link

Comment (benesch) [inline releases.md]: Done!

**Comment by benesch** [comment]
@sploiselle, want me to Title Case all the column headings? (Also, did you want "Version & Release Notes" instead of "Version and Release Notes"?)

**Comment by sploiselle** [comment]
I think most everything is titled cased, but I can't find a table header example easily, so I'll defer to @jseldess.

I prefer ampersands because they take up less real estate on the screen.

**Comment by benesch** [comment]
It's title-cased in [your backup documentation](https://www.cockroachlabs.com/docs/backup.html#object-dependencies), at least ;). Also in [Build an App with CockroachDB}(https://www.cockroachlabs.com/docs/build-an-app-with-cockroachdb.html).

Sending up a fix in a hot second.

**Comment by jseldess** [comment]
Wohoo! This is very nice. Yes, let's title-case the column-headings, and I like `&` for the same reason as @sploiselle.

It'd be great if we could hold off on merging this until just after https://github.com/cockroachdb/docs/pull/1357, which will mean some minor rebasing here.

File: releases.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
+---
+title: Releases
+summary: Release notes for older versions of CockroachDB.
+toc: false
+---
+
+<style>
```
Comment (jseldess) [inline releases.md]: To separate the content from the styling, how about adding this as `/css/components/_release.scss` and then adding `@import "components/release";` to `/css/customstyles.scss`?

**Comment by jseldess** [comment]
@benesch, free to merge now, once you've addressed any open issues.

**Comment by benesch** [comment]
Ok, I think I've addressed everything! Merging as soon as this goes green.

**Comment by jseldess** [comment]
@benesch, not sure what we missed, but the published version of the page has some styling issue: 

- The latest tag is not colored right.
- The columns aren't spaced well.

Ideas? 

screen shot 2017-05-05 at 12 25 37 pm

**Comment by benesch** [comment]
Hmm, it seems okay on my machine. My hunch is you're picking up a cached version of `customstyles.css`. Try clearing your cache or incognito mode?

**Comment by benesch** [comment]
Turned out to be caching, luckily!

**Comment by jseldess** [comment]
Yep. Thanks, @benesch!

**Pull request merged**